src/functions.jl

```diff
@@ -6,649 +6,479 @@
+        end
 
-## Log ## 
-doc"""
-    log(a)
+        ## log ##
+        function log(a::$T)
+            constant_term(a) == zero(constant_term(a)) &&
+                throw(ArgumentError("""
+                    The 0-th order `TaylorN` coefficient must be non-zero
+                    in order to expand `log` around 0.
+                    """))
+
+            order = max_order(a)
+            @inbounds aux = log( constant_term(a) )
+            c = $T( aux, order )
+            @inbounds for k = 1:order
+                log!(c, a, k)
+            end
+            return c
+        end
 
-Return the Taylor expansion of $\log(a)$, of order `a.order`, for `a::Taylor1`
-polynomial.
+        ## sin and cos ##
+        sin(a::$T) = sincos(a)[1]
+        cos(a::$T) = sincos(a)[2]
+        function sincos(a::$T)
+            order = max_order(a)
+            @inbounds s = $T( sin(constant_term(a)), order )
+            @inbounds c = $T( cos(constant_term(a)), order )
+            @inbounds for k = 1:order
+                sincos!(s, c, a, k)
+            end
+            return s, c
+        end
 
-For details on making the Taylor expansion, see
-[`TaylorSeries.logHomogCoef`](@ref).
-"""
-function log(a::Taylor1)
-    findfirst(a)>0 && throw(
-        ArgumentError("Impossible to expand `log` around 0."))
-    @inbounds aux = log( a[1] )
-    T = typeof(aux)
-    ac = convert(Array{T,1}, a.coeffs)
-    coeffs = similar(ac)
-    @inbounds coeffs[1] = aux
-    @inbounds for k = 1:a.order
-        coeffs[k+1] = logHomogCoef(k, ac, coeffs)
-    end
-    Taylor1( coeffs, a.order )
-end
+        ## tan ##
+        function tan(a::$T)
+            order = max_order(a)
+            @inbounds aux = tan(constant_term(a))
+            c = $T(aux, order)
+            c2 = $T(aux^2, order)
+            @inbounds for k = 1:order
+                tan!(c, a, c2, k)
+            end
+            return c
+        end
 
-# Homogeneous coefficients for log
-doc"""
-    logHomogCoef(kcoef, ac, coeffs)
+        ## asin ##
+        function asin(a::$T)
+            a0 = constant_term(a)
+            a0^2 == one(a0) && throw(ArgumentError(
+                """
+                Recursion formula diverges due to vanishing `sqrt`
+                in the denominator.
+                """))
+
+            order = max_order(a)
+            c = $T( asin(a0), order )
+            r = $T( sqrt(1 - a0^2), order )
+            @inbounds for k in 1:order
+                asin!(c, a, r, k)
+            end
+            return c
+        end
 
-Compute the `k-th` expansion coefficient of $c = \log(a)$, given by
+        ## acos ##
+        function acos(a::$T)
+            a0 = constant_term(a)
+            a0^2 == one(a0) && throw(ArgumentError(
+                """
+                Recursion formula diverges due to vanishing `sqrt`
+                in the denominator.
+                """))
+
+            order = max_order(a)
+            c = $T( asin(a0), order )
```

Comment:
It's actually a nice trick that @blas-ko came up with: the recursion formula is essentially the same as for `asin` except for a sign and the constant term.